notebooks/02-ingestion/12-spatial-dimensions.ipynb

```diff
@@ -0,0 +1,709 @@
+    "        if result[\"status\"][\"status\"] != 'RUNNING':\n",
+    "            done = True\n",
+    "        else:\n",
+    "            probe_count+=1\n",
+    "            print(f'Sleeping... probe # {probe_count}')\n",
+    "            time.sleep(1)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "9464c204-bbf1-4a94-ae02-2356b606049f",
+   "metadata": {},
+   "source": [
+    "## Generate sample data\n",
+    "The following cell creates 10k rows of user data that include latitude and longitude coordinates for each user."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "e9f69c6e-ccf2-46de-b7c6-2cfebd590017",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# simulate clicks for last 2 hours\n",
+    "gen_hours=2\n",
+    "gen_now = datetime.now() - timedelta(hours=gen_hours)\n",
+    "gen_start_time = gen_now.strftime(\"%Y-%m-%d %H:%M:%S\")\n",
+    "\n",
+    "headers = {\n",
+    "  'Content-Type': 'application/json'\n",
+    "}\n",
+    "\n",
+    "datagen_request = {\n",
+    "    \"name\": \"users\",\n",
+    "    \"target\": { \"type\": \"file\", \"path\":\"users.json\"  },\n",
+    "    \"config_file\": \"clickstream/users_init.json\", \n",
+    "    \"concurrency\":100,\n",
+    "    \"total_events\":10000 \n",
+    "}\n",
+    "datagen.post(\"/start\", json.dumps(datagen_request), headers=headers)\n",
+    "wait_for_datagen('users')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "1e312f34-313f-4eb4-9d94-98682f09eb05",
+   "metadata": {},
+   "source": [
+    "To ingest the latitude and longitude as a spatially indexed dimension, you must currently use a native ingestion spec. SQL Based ingestion does not yet support spatial dimensions.\n",
+    "\n",
+    "The native `index_parallel` spec below includes a property within the \"dimensionSpec\" that describes the spatial index creation:\n",
+    "```\n",
+    "    \"dimensionSpec\":\n",
+    "      \"dimensions\":[\n",
+    "          ...\n",
+    "          {\n",
+    "            \"name\": \"address_lat\",\n",
+    "            \"type\": \"double\"\n",
+    "          },\n",
+    "          {\n",
+    "            \"name\": \"address_long\",\n",
+    "            \"type\": \"double\"\n",
+    "          },\n",
+    "          ...\n",
+    "        ]\n",
+    "      \"spatialDimensions\": [\n",
+    "          {\n",
+    "             \"dimName\": \"address_spatial\",\n",
+    "             \"dims\": [\n",
+    "               \"address_lat\",\n",
+    "               \"address_long\"\n",
+    "             ]\n",
+    "          }\n",
+    "        ]\n",
+    "       ...\n",
+    "```\n",
+    "\n",
+    "The \"spatialDimensions\" array contains the list of spatial dimensions, each spatial dimension has a name (\"dimName\") and the list of dimensions that are the coordinates for a point in space. The dimensions that form the point come from the dimensions defined in the ingestion spec.\n",
```

Comment:
Use code font for spatialDimensions and dimName

Reply:
done